### PR TITLE
Update Canal CNI to v3.10

### LINKED ISSUE #599: Update Canal to the latest version
**What feature would you like to be added?**

We haven't updated Canal (Calico + Flannel) for a long time and we should try to get it updated.

**What are use cases of the feature?**

The new version brings new features, such as integration with WireGuard, making Canal a CNI plugin that supports encryption.

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

Update Canal CNI to v3.10.

Docs: https://docs.projectcalico.org/v3.10/getting-started/kubernetes/installation/flannel
Manifest: https://docs.projectcalico.org/v3.10/manifests/canal.yaml

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #599 

**Special notes for your reviewer**:

Blocked until upgrade E2E tests are not fixed (#717).
/hold

**Does this PR introduce a user-facing change?**:
```release-note
Update Canal CNI to v3.10
```

### CODE REVIEW

**Comment by xmudrii** [comment]
/assign @kron4eg 
The PR is ready for review, but I'll leave it on hold until I don't finish testing it manually

File: pkg/templates/canal/prerequisites.go [hunk 1]
```diff
@@ -210,6 +216,7 @@ func calicoClusterRole() *rbacv1.ClusterRole {
 	}
 }
 
+// TODO(xmudrii): This doesn't exist anymore in the original manifest.
```
Comment (xmudrii) [inline pkg/templates/canal/prerequisites.go]: @kron4eg That ClusterRole is not present in the manifest anymore and I don't know who creates that ServiceAccount. Should we drop this role?

**Comment by xmudrii** [comment]
Timeout error
/retest

**Comment by xmudrii** [comment]
This doesn't affect upgrades because we never upgrade CNI. I'll create a follow-up PR that fixes this, but in meanwhile we can merge this one.
/hold cancel

File: pkg/templates/canal/daemonset.go [hunk 2]
```diff
@@ -66,8 +67,11 @@ func daemonSet(ifacePatch bool) *appsv1.DaemonSet {
 		flannelEnv = append(flannelEnv, corev1.EnvVar{
 			Name: "FLANNELD_IFACE",
 			ValueFrom: &corev1.EnvVarSource{
-				FieldRef: &corev1.ObjectFieldSelector{
-					FieldPath: "status.hostIP",
+				ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
+					Key: "canal_iface",
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: "canal-config",
+					},
```
Comment (kron4eg) [inline pkg/templates/canal/daemonset.go]: this change will break cluster with `external: true`

Comment (xmudrii) [inline pkg/templates/canal/daemonset.go]: I believe this should be reverted

Comment (kron4eg) [reply pkg/templates/canal/daemonset.go]: see cb97908a3825e72484b15ed446216e17d851c001 for more info

Comment (xmudrii) [reply pkg/templates/canal/daemonset.go]: @kron4eg I've already reverted it, please see the latest commit.

File: pkg/templates/canal/prerequisites.go (hunk 1 above)
Comment (kron4eg) [reply pkg/templates/canal/prerequisites.go]: Yeah, probably yes.

Yet we don't have effective mechanism to delete "old stuff" currently.  Something like `kubectl apply --prune -l app=kubeone` would do. Probably we need to work on implementing it.

**Pull request merged**